modules/grandpa/src/lib.rs

```diff
@@ -819,9 +835,26 @@ mod tests {
 	fn succesfully_imports_header_with_valid_finality() {
 		run_test(|| {
 			initialize_substrate_bridge();
-			let result = submit_finality_proof(1);
+
+			let header_number = 1;
+			let header = test_header(header_number.into());
+			let justification = make_default_justification(&header);
+
+			let pre_dispatch_weight = <TestRuntime as Config>::WeightInfo::submit_finality_proof(
+				justification.commit.precommits.len().try_into().unwrap_or(u32::MAX),
+				justification.votes_ancestries.len().try_into().unwrap_or(u32::MAX),
+			);
+
+			let result = submit_finality_proof(header_number);
 			assert_ok!(result);
 			assert_eq!(result.unwrap().pays_fee, frame_support::dispatch::Pays::Yes);
+			// our test config assumes 2048 max authorities and we are just using couple
+			let pre_dispatch_proof_size = pre_dispatch_weight.proof_size();
+			let actual_proof_size = result.unwrap().actual_weight.unwrap().proof_size();
+			assert!(
+				actual_proof_size < pre_dispatch_proof_size,
```

Comment:
I have the following values:

3 precommits
2 votes ancestries

pre_dispatch_proof_size: 4745
The number of authorities in the previous set: 3. So extra authorities: 2045
extra_proof_size_bytes: 40 * 2045 = 81800
After saturated subtraction, actual_proof_size: 0

Does this seem correct ? Just checking. The difference seems very big.

Reply:
Yes, unfortunately that's a mismatch between Millau runtime and mocked runtime. So on Millau we have `MaxBridgedAuthorities` set to `5` (which is a correct value). This runtime is used to generate weights and weights include that `proof_size`. In mock runtime we have the `MaxBridgedAuthorities` set to `2048`. So when we run tests, we use weights, generated using `MaxBridgedAuthorities = 5`, but we are executing in the runtime that uses `MaxBridgedAuthorities = 2048`. That's why the difference is that large. I can change `MaxBridgedAuthorities` to `5` in mock runtime if you think it'll be better. But the main thing is that the `proof_size` gets refunded - so `actual_proof_size < pre_dispatch_proof_size`, no matter the refund amount. For amount there's an other test - `extra_proof_size_bytes_works`

Reply:
I see. Yes, if it doesn't affect anything else, I think it would be better to change `MaxBridgedAuthorities` to 5 in the mock and maybe also test that `actual_proof_size > 0`.